server/sonar-server/src/main/java/org/sonar/server/platform/ws/SystemInfoWriter.java

```diff
@@ -43,33 +38,13 @@
     "Compute Engine Tasks", "Compute Engine JVM State", "Compute Engine Database Connection", "Compute Engine Logging", "Compute Engine JVM Properties",
     "Search State", "Search Indexes"};
 
-  private final UserSession userSession;
   private final TelemetryDataLoader telemetry;
 
-  public BaseInfoWsAction(UserSession userSession, TelemetryDataLoader telemetry) {
-    this.userSession = userSession;
+  SystemInfoWriter(TelemetryDataLoader telemetry) {
     this.telemetry = telemetry;
   }
 
-  @Override
-  public void define(WebService.NewController controller) {
-    controller.createAction("info")
-      .setDescription("Get detailed information about system configuration.<br/>" +
-        "Requires 'Administer' permissions.<br/>" +
-        "Since 5.5, this web service becomes internal in order to more easily update result.")
-      .setSince("5.1")
-      .setInternal(true)
-      .setResponseExample(getClass().getResource("/org/sonar/server/platform/ws/info-example.json"))
-      .setHandler(this);
-  }
-
-  @Override
-  public void handle(Request request, Response response) throws InterruptedException {
-    userSession.checkIsSystemAdministrator();
-    doHandle(request, response);
-  }
-
-  protected abstract void doHandle(Request request, Response response) throws InterruptedException;
+  public abstract void write(JsonWriter json) throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)